chain-impl-mockchain/src/testing/builders/tx_cert_builder.rs

```diff
@@ -200,17 +225,58 @@ pub fn tally_sign(
 ) -> TallyProof {
     let payload_type = vt.tally_type();
 
+    let key: EitherEd25519SecretKey = keys[0].clone();
+    let id = key.to_public().into();
+
+    let auth_data = builder.get_auth_data();
+    let signature = SingleAccountBindingSignature::new(&auth_data, |d| key.sign_slice(&d.0));
+
+    match payload_type {
+        PayloadType::Public => TallyProof::Public { id, signature },
+        PayloadType::Private => unreachable!("Vote tally payload should never be private"),
+    }
+}
+
+pub fn encrypted_tally_sign(
+    keys: &[EitherEd25519SecretKey],
+    vt: &EncryptedVoteTally,
+    builder: &TxBuilderState<SetAuthData<EncryptedVoteTally>>,
+) -> EncryptedVoteTallyProof {
+    let payload_type = vt.tally_type();
+
+    let key: EitherEd25519SecretKey = keys[0].clone();
+    let id = key.to_public().into();
+
+    let auth_data = builder.get_auth_data();
+    let signature = SingleAccountBindingSignature::new(&auth_data, |d| key.sign_slice(&d.0));
+
     match payload_type {
-        PayloadType::Public => {
-            let key: EitherEd25519SecretKey = keys[0].clone();
-            let id = key.to_public().into();
+        PayloadType::Public => unreachable!("Encrypted vote tally payload should never be public"),
+        PayloadType::Private => EncryptedVoteTallyProof { id, signature },
+    }
+}
 
-            let auth_data = builder.get_auth_data();
-            let signature =
-                SingleAccountBindingSignature::new(&auth_data, |d| key.sign_slice(&d.0));
+pub fn private_tally_sign(
+    keys: &[EitherEd25519SecretKey],
+    vt: &PrivateVoteTally,
+    builder: &TxBuilderState<SetAuthData<PrivateVoteTally>>,
+) -> TallyProof {
+    let payload_type = vt.tally_type();
 
-            TallyProof::Public { id, signature }
-        }
+    let key: EitherEd25519SecretKey = keys[0].clone();
+    let id = key.to_public().into();
+
+    let auth_data = builder.get_auth_data();
+    let signature = SingleAccountBindingSignature::new(&auth_data, |d| key.sign_slice(&d.0));
+
+    match payload_type {
+        PayloadType::Public => unreachable!("Encrypted vote tally payload should never be public"),
+        PayloadType::Private => TallyProof::Private {
+            id,
+            signature,
+            // TODO: how to get the shares here?
+            shares: vec![],
```

Comment:
So, how to have the shares within the private payload?

Reply:
I think this function should just accept them as bytes. Is there a problem with doing that?